docx/oxml/section.py

```diff
@@ -41,22 +45,24 @@ class CT_SectPr(BaseOxmlElement):
     """
     ``<w:sectPr>`` element, the container element for section properties.
     """
-    __child_sequence__ = (
-        'w:footnotePr', 'w:endnotePr', 'w:type', 'w:pgSz', 'w:pgMar',
-        'w:paperSrc', 'w:pgBorders', 'w:lnNumType', 'w:pgNumType', 'w:cols',
-        'w:formProt', 'w:vAlign', 'w:noEndnote', 'w:titlePg',
+    _tag_seq = (
+        'w:headerReference', 'w:footnotePr', 'w:endnotePr', 'w:type', 'w:pgSz',
+        'w:pgMar', 'w:paperSrc', 'w:pgBorders', 'w:lnNumType', 'w:pgNumType',
+        'w:cols', 'w:formProt', 'w:vAlign', 'w:noEndnote', 'w:titlePg',
         'w:textDirection', 'w:bidi', 'w:rtlGutter', 'w:docGrid',
         'w:printerSettings', 'w:sectPrChange',
     )
+    headerReference = ZeroOrMore('w:headerReference', successors=())
     type = ZeroOrOne('w:type', successors=(
-        __child_sequence__[__child_sequence__.index('w:type')+1:]
+        _tag_seq[_tag_seq.index('w:type')+1:]
     ))
     pgSz = ZeroOrOne('w:pgSz', successors=(
-        __child_sequence__[__child_sequence__.index('w:pgSz')+1:]
+        _tag_seq[_tag_seq.index('w:pgSz')+1:]
     ))
     pgMar = ZeroOrOne('w:pgMar', successors=(
-        __child_sequence__[__child_sequence__.index('w:pgMar')+1:]
+        _tag_seq[_tag_seq.index('w:pgMar')+1:]
     ))
+    del _tag_seq
```

Comment:
We distinguish here between child elements and attributes of the current element. `_tag_seq` enumerates the possible child elements, in sequence, just to provide a compact way to specify successor elements for any given child tag/element.

So ZeroOrOne() is logically ZeroOrOneChildElementHavingThisTag() whereas Optional/RequiredAttribute define an attribute on this element, rather than a child element.

As far as what goes before the `del _tag_seq`, just have a look around and make it consistent with the first couple others you find that show a similar situation. I don't remember off the top of my head what convention I've adopted. I do note that elements seem to be kind of one or the other, i.e. have children or have attributes, but I expect there are a couple cases around :)

Both ZeroOrOne and Optional reflect optionality, like minOccurs="0" or use="optional". Optionality is a bit different on attributes since the most they can appear is once where a child element can be repeating.